graylog2-web-interface/src/components/search/MessageDetail.tsx

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import React from 'react';
+import Immutable from 'immutable';
+
+import { Link } from 'components/common/router';
+import { MessageDetailsDefinitionList, ClipboardButton, Icon, Timestamp } from 'components/common';
+import { ButtonGroup, Col, Label, Row } from 'components/bootstrap';
+import StreamLink from 'components/streams/StreamLink';
+import MessageFields from 'components/search/MessageFields';
+import MessageDetailsTitle from 'components/search/MessageDetailsTitle';
+import Routes from 'routing/Routes';
+import AppConfig from 'util/AppConfig';
+import MessagePermalinkButton from 'views/components/common/MessagePermalinkButton';
+import type { Message } from 'views/components/messagelist/Types';
+import type { Stream } from 'views/stores/StreamsStore';
+import type { Input } from 'components/messageloaders/Types';
+
+const MessageActions = ({ messageIndex, messageId }: { messageIndex: string | undefined, messageId: string }) => (
+  <ButtonGroup className="pull-right">
+    <MessagePermalinkButton messageIndex={messageIndex} messageId={messageId} />
+
+    <ClipboardButton title="Copy ID" bsSize="small" text={messageId} />
+  </ButtonGroup>
+);
+
+const InputName = ({ inputs, inputId }: { inputs: Immutable.Map<string, Input> | undefined, inputId: string }) => {
+  const input = inputs?.get(inputId);
+
+  return input ? <span style={{ wordBreak: 'break-word' }}>{input.title}</span> : <>deleted input</>;
```

Comment:
I think we should extract a styled component for all the `span`'s with the `word-break` property.